packages/codec/lib/wrap/priority.ts

```diff
@@ -52,6 +60,13 @@ export function isMoreSpecific(
     ["string"],
     ["bool"]
   ];
+  if (strictBooleans) {
+    //swap priority of strings and bools
+    typeClasses.pop();
+    typeClasses.pop();
```

Comment:
This looks brittle; can we do it another way?

Reply:
Sure, there's any number of alternative ways to do it, the question is what's good without being too repetitive.  I guess we could have everything but the last two, then tack on the last two depending on the flag?  Would that be less brittle?

Reply:
(I've made this change, let me know if you think a different way would be better.)